packages/@blockprotocol/graph/src/stdlib.ts

```diff
@@ -96,15 +96,22 @@ export const getRightEntityForLinkEntity = (
   entityId: EntityId,
 ): Entity => getRightEntityForLinkEntityTemporal(subgraph, entityId);
 
+type GenericLinkEntityAndRightEntity = {
+  linkEntity: Entity;
+  rightEntity: Entity;
+};
+
 /**
  * Gets all outgoing link entities, and their "target" entities (by default this is the "right entity"), from a given
  * entity.
  *
  * @param subgraph
  * @param {EntityId} entityId - The ID of the source entity to search for outgoing links from
  */
-export const getOutgoingLinkAndTargetEntities = (
+export const getOutgoingLinkAndTargetEntities = <
+  LinkData extends GenericLinkEntityAndRightEntity[] = GenericLinkEntityAndRightEntity[],
+>(
   subgraph: Subgraph,
   entityId: EntityId,
-): { linkEntity: Entity; rightEntity: Entity }[] =>
-  getOutgoingLinkAndTargetEntitiesTemporal(subgraph, entityId);
+): LinkData =>
+  getOutgoingLinkAndTargetEntitiesTemporal(subgraph, entityId) as LinkData; // @todo add generics elsewhere
```

Comment:
Just adding them to more functions down the chain to avoid casting here – I started to do this in the next PR but stopped again, let's discuss in follow up